back-end/data_controller.rb

```diff
@@ -1,11 +1,33 @@
 class DataController < ApplicationController
-  def validate_params
+  TEMPERATURE = 'temperature'.freeze 
+  NOISE = 'noise'.freeze
+  PRESSURE = 'pressure'.freeze
+  HUMIDITY = 'humidity'.freeze
+  LIGHT = 'light'.freeze
+
+  VALID_METRICS = [TEMPERATURE, NOISE, PRESSURE, HUMIDITY, LIGHT].freeze
+
+  def valid_params?(metrics)
+    metrics.each do |metric_key|
+      possible_metric = metrics[metric_key]
+      return false if !VALID_METRICS.include?(possible_metric)
+    end
+  end
+
+  def valid_timestamp?(start_time, stop_time)
+    return true if start_time <= Time.Now && stop_time > Time.Now
```

Comment:
start and stop times can both be in the past.  Also, we should check that the start time is earlier than the stop time.